homeassistant/components/matter/update.py

```diff
@@ -0,0 +1,225 @@
+    """Extra stored data for Matter node firmware update entity."""
+
+    software_update: MatterSoftwareVersion | None = None
+
+    def as_dict(self) -> dict[str, Any]:
+        """Return a dict representation of the extra data."""
+        return {
+            ATTR_SOFTWARE_UPDATE: self.software_update.as_dict()
+            if self.software_update is not None
+            else None,
+        }
+
+    @classmethod
+    def from_dict(cls, data: dict[str, Any]) -> MatterUpdateExtraStoredData:
+        """Initialize the extra data from a dict."""
+        if data[ATTR_SOFTWARE_UPDATE] is None:
+            return cls()
+        return cls(MatterSoftwareVersion.from_dict(data[ATTR_SOFTWARE_UPDATE]))
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Matter lock from Config Entry."""
+    matter = get_matter(hass)
+    matter.register_platform_handler(Platform.UPDATE, async_add_entities)
+
+
+class MatterUpdate(MatterEntity, UpdateEntity):
+    """Representation of a Matter node capable of updating."""
+
+    _software_update: MatterSoftwareVersion | None = None
+    _cancel_update: CALLBACK_TYPE | None = None
+
+    @callback
+    def _update_from_device(self) -> None:
+        """Update from device."""
+
+        self._attr_installed_version = self.get_matter_attribute_value(
+            clusters.BasicInformation.Attributes.SoftwareVersionString
+        )
+
+        if self.get_matter_attribute_value(
+            clusters.OtaSoftwareUpdateRequestor.Attributes.UpdatePossible
+        ):
+            self._attr_supported_features = (
+                UpdateEntityFeature.INSTALL
+                | UpdateEntityFeature.PROGRESS
+                | UpdateEntityFeature.SPECIFIC_VERSION
+            )
+
+        update_state: clusters.OtaSoftwareUpdateRequestor.Enums.UpdateStateEnum = (
+            self.get_matter_attribute_value(
+                clusters.OtaSoftwareUpdateRequestor.Attributes.UpdateState
+            )
+        )
+        if (
+            update_state
+            == clusters.OtaSoftwareUpdateRequestor.Enums.UpdateStateEnum.kIdle
+        ):
+            self._attr_in_progress = False
+            return
+
+        update_progress: int = self.get_matter_attribute_value(
+            clusters.OtaSoftwareUpdateRequestor.Attributes.UpdateStateProgress
+        )
+
+        if (
+            update_state
+            == clusters.OtaSoftwareUpdateRequestor.Enums.UpdateStateEnum.kDownloading
+            and update_progress is not None
+            and update_progress > 0
+        ):
+            self._attr_in_progress = update_progress
+        else:
+            self._attr_in_progress = True
+
+    async def async_update(self) -> None:
+        """Call when the entity needs to be updated."""
+        try:
+            update_information = await self.matter_client.check_node_update(
```

Comment:
Does this send a message to the Matter devices too or is it just a local update check?

If the update button on the "Updates" page or the "Check for updates" menu button on the HA settings page is clicked, `async_update` is called for all update entities.
If an "image notify" message is sent to each device (e.g. via Thread) here, this could cause the network to be spammed for a short amount of time.

So, if this sends a message to the device, maybe set `PARALLEL_UPDATES` to `1` or another reasonable number for the Matter update entity/platform?

Reply:
> Does this send a message to the Matter devices too or is it just a local update check?

This causes no communication with devices. It uses cached vendor/product ID information to check for updates. The update check is polling the http get requests to the CSA DCL REST API.